python/llm/src/bigdl/llm/transformers/convert_ipex.py

```diff
@@ -91,18 +95,47 @@ def _ipex_optimize_attention(model):
             supported_mha_class,
             _IPEXAttentionCPU,
             model.config,
-            tpp=True if _using_tpp() else False,
-            woq=False,
+            tpp=is_tpp,
+            woq=is_woq,
         )
 
 
-def _ipex_optimize_model(model, rms_classes):
-    _enable_tpp()
+def _ipex_optimize_model(model, rms_classes, qtype):
     import intel_extension_for_pytorch as ipex
-    ipex.optimize(model.eval(), dtype=torch.bfloat16, inplace=True).eval()
-    _ipex_optimize_rmsnorm(model, rms_classes)
-    _ipex_optimize_attention(model)
-    _ipex_optimize_decoder(model)
+    from intel_extension_for_pytorch.transformers.models.reference.models import output_hook
+    from intel_extension_for_pytorch.transformers.optimize import ipex_quantization_flow
+
+    is_woq = False
+    is_quantization = False
+    _disable_tpp()
+    if qtype == ggml_tensor_qtype["bf16"]:
+        _enable_tpp()
+        model = ipex.optimize(model.eval(), dtype=torch.bfloat16, inplace=True).eval()
+    elif qtype == ggml_tensor_qtype["sym_int4"]:
+        is_quantization = True
+        is_woq = True
+        act_quant_mode_dict = {
+            "PER_TENSOR": ipex.quantization.WoqActQuantMode.PER_TENSOR,
+            "PER_IC_BLOCK": ipex.quantization.WoqActQuantMode.PER_IC_BLOCK,
+            "PER_BATCH": ipex.quantization.WoqActQuantMode.PER_BATCH,
+            "PER_BATCH_IC_BLOCK": ipex.quantization.WoqActQuantMode.PER_BATCH_IC_BLOCK,
+        }
+        qconfig = ipex.quantization.get_weight_only_quant_qconfig_mapping(
+            weight_dtype=torch.quint4x2,  # INT4
+            lowp_mode=ipex.quantization.WoqLowpMode.INT8,
+            act_quant_mode=act_quant_mode_dict["PER_IC_BLOCK"],
+            group_size=-1,
+        )
+        model = ipex_quantization_flow(model, torch.bfloat16, None, qconfig, None)
```

Comment:
Only llama is verified, yes?

Reply:
Ziteng and I has verified llama2, vicuna, chatglm3, baichuan2, qwen, mistral, starcoder and ziya model.